partials/index/Standard.njk

```diff
@@ -0,0 +1,179 @@
+    try{
+      {%- if asyncapi | containsBinaryPayload %}
+      this.options.payload = Payload.BINARY;
+      this.binaryClient = await connect(this.options);
+      this.chainEvents(this.binaryClient);
+      {%- endif %}
+      {%- if asyncapi | containsStringPayload %}
+      this.options.payload = Payload.STRING;
+      this.stringClient = await connect(this.options);
+      this.chainEvents(this.stringClient);
+      {%- endif %}
+      {%- if asyncapi | containsJsonPayload %}
+      this.options.payload = Payload.JSON;
+      this.jsonClient = await connect(this.options);
+      this.chainEvents(this.jsonClient);
+      {%- endif %}
+    }catch(e){
+      this.emit(AvailableEvents.connectionError, e)
+    }
+  }
+
+
+  /**
+   * Disconnect all clients from the server
+   */
+  async disconnect(){
+    {%- if asyncapi | containsBinaryPayload %}
+    this.binaryClient!.close()
+    {%- endif %}
+    {%- if asyncapi | containsStringPayload %}
+    this.stringClient!.close()
+    {%- endif %}
+    {%- if asyncapi | containsJsonPayload %}
+    this.jsonClient!.close()
+    {%- endif %}
+  }
+  
+  private chainEvents(ns: Client){
+      ns.on('permissionError', (e: NatsError) => {
+        this.emit(AvailableEvents.permissionError, e)
+      });
+      ns.on('close', (e: NatsError) => {
+        this.emit(AvailableEvents.close, e)
+      });
+      ns.on('connect', (connection: Client, serverURL: string, info: ServerInfo) => {
+        this.emit(AvailableEvents.connect, connection, serverURL, info)
+      });
+      ns.on('connecting', (serverURL: NatsError) => {
+        this.emit(AvailableEvents.connecting, serverURL)
+      });
+      ns.on('disconnect', (serverURL: string) => {
+        this.emit(AvailableEvents.disconnect, serverURL)
+      });
+      ns.on('error', (e: NatsError) => {
+        this.emit(AvailableEvents.error, e)
+      });
+      ns.on('pingcount', () => {
+        this.emit(AvailableEvents.pingcount)
+      });
+      ns.on('pingtimer', () => {
+        this.emit(AvailableEvents.pingtimer)
+      });
+      ns.on('reconnect', (connection: Client, serverURL: string, info: ServerInfo) => {
+        this.emit(AvailableEvents.reconnect, connection, serverURL, info)
+      });
+      ns.on('reconnecting', (serverURL: string) => {
+        this.emit(AvailableEvents.reconnecting, serverURL)
+      });
+      ns.on('serversChanged', (e: ServersChangedEvent) => {
+        this.emit(AvailableEvents.serversChanged, e)
+      });
+      ns.on('subscribe', (e: SubEvent) => {
+        this.emit(AvailableEvents.subscribe, e)
+      });
+      ns.on('unsubscribe', (e: SubEvent) => {
+        this.emit(AvailableEvents.unsubscribe, e)
+      });
+      ns.on('yield', () => {
+        this.emit(AvailableEvents.yield)
+      });
+  }
+  /**
+   * Try to connect to the NATS server with nkey authentication
+   */
+  async connectWithNkey(options : NatsConnectionOptions, publicNkey: string, seed: string){
+    options.nkey = publicNkey;
```

Comment:
Depending on the library, the nonceSigner will be generated for you. For sure if you specify as creds file, the nonce is generated on master. On v2, the client generates the signer function for you as well if the JWT and nkey are provided